### PR TITLE
Create CMake build for x64 and x32

### PR DESCRIPTION
**Pull request by galanCA**
## Description

Replace the content in this section with:
- The motivation and context for this change (if it is not immediately clear from the title)
- If it fixes an open issue, specify the issue number (e.g., "fixes #XXXX")
- A summary of the behavior expected from this change
- A description of tests performed

## Author Progress Checklist:

- [ ] Open draft pull request
    - [ ] Make title clearly understandable in a standalone change log context
    - [ ] Assign yourself the issue
    - [ ] Add at least one label (enhancement, bug, or maintenance)
    - [ ] Link the issue(s) addressed by this PR (under "Development" in the sidebar menu) 
- [ ] Make code changes (if you haven't already)
- [ ] Self-review of code
    - [ ] My code follows the style guidelines of this project
    - [ ] I have added comments to my code, particularly in hard-to-understand areas
    - [ ] I have only committed the necessary changes for this fix or feature
    - [ ] I have made corresponding changes to the documentation
    - [ ] My changes generate no new warnings
    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
        - [ ] exercising the code changes in the test framework, and/or
        - [ ] manually verifying the changes (as explained in the the pull request description above)
    - [ ] My changes pass all local tests
    - [ ] My changes successfully passes CI checks
    - [ ] Add any unit test for proof and documentation.
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
- [ ] Move pull request out of draft mode and assign reviewers
- [ ] Iterate with reviewers until all changes are approved
    - [ ] Make changes in response to reviewer comments
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
    - [ ] Re-request review in GitHub

## Reviewer Checklist:

 - [ ] Read the pull request description
 - [ ] Perform a code review on GitHub
 - [ ] Confirm all CI checks pass and there are no build warnings
 - [ ] Pull, build, and run automated tests locally
 - [ ] Perform manual tests of the fix or feature locally
 - [ ] Add any review comments, if applicable
 - [ ] Submit review in GitHub as either
     - [ ] Request changes, or
     - [ ] Approve
 - [ ] Iterate with author until all changes are approved

### CODE REVIEW

File: cmake/BuildFlags.cmake [hunk 1]
```diff
@@ -52,7 +52,12 @@ target_compile_options(cse_common_interface INTERFACE
     /nologo       # Suppresses display of sign-on banner.
     /fp:precise   #+Specifies floating-point behavior.
     /fp:except-   # Specifies floating-point behavior.
-    /arch:IA32    # Specifies the architecture for code generation (no special instructions).
+    $<$<STREQUAL:"${CSEARCHBUILDFLAG}","Win32">:
+      /arch:IA32    # Specifies the architecture for code generation (no special instructions).
+    >
+    $<$<STREQUAL:"${CSEARCHBUILDFLAG}","x64">:
+      /arch:AVX    # Enables the use of Intel Advanced Vector Extensions instructions.
```
Comment (galanCA) [inline cmake/BuildFlags.cmake:59]: @nealkruis Is this an ok flag to replace IA32 when using x64? https://learn.microsoft.com/en-us/cpp/build/reference/arch-x64?view=msvc-170

Another option is to leave it empty which uses the flag SSE2. Let me know what you think about it.

**Comment by nealkruis** [comment]
@galanCA this PR has test errors. Looks similar to the SDK errors again, but not sure if it's related.

**Comment by galanCA** [comment]
Yes, I am already looking into it.

**Comment by galanCA** [comment]
@nealkruis The problem was the quote around `${BUILD_ARCHITECTURE}`.  I am still trying to figure out why that is the case.

**Comment by galanCA** [comment]
Adding this link to show the reason to comment out SSE2. https://stackoverflow.com/questions/1067630/sse2-option-in-visual-c-x64

**Review by nealkruis — APPROVED** [review]

**Pull request merged**